python/cudf/cudf/_lib/pylibcudf/lists.pyx

```diff
@@ -94,3 +100,29 @@ cpdef Column concatenate_list_elements(Column input, bool dropna):
         ))
 
     return Column.from_libcudf(move(c_result))
+
+
+cpdef Column extract_list_element(Column input, ColumnOrSizeType index):
+    """Create a column of extracted list elements.
+    Parameters
+    ----------
+    input : Column
+        The input column.
+    index : Union[Column, size_type]
+        The selection index or indices.
+    Returns
+    -------
+    Column
+        A new Column of bools
+    """
+    cdef unique_ptr[column] c_result
+    cdef shared_ptr[lists_column_view] list_view = (
+        make_shared[lists_column_view](input.view())
+    )
+    if ColumnOrSizeType is Column:
+        with nogil:
+            c_result = move(cpp_extract_list_element(list_view.get()[0], index.view()))
+    else:
+        with nogil:
+            c_result = move(cpp_extract_list_element(list_view.get()[0], index))
```

Comment:
*issue*: I think unfortunately, we need to handle the case that the argument is `object` (see here https://github.com/rapidsai/cudf/pull/15257#discussion_r1610591457)

@vyasr I think we should put something in the dev guide.

Reply:
Yes, we should. I will also try and set aside a bit of time to investigate fixing this upstream. It's quite annoying how far we're having to propagate this workaround.

Reply:
[`size_type`](https://docs.rapids.ai/api/libcudf/stable/group__utility__types#ga0aff16b2db009d92300befaca11827e9) is not a `cpdef enum class`. So we wouldn't need to check for `object`, right?